P2PEnabledConference.js

```diff
@@ -0,0 +1,726 @@
+            jingleSession.isP2P = !this.room.isFocus(jingleSession.peerjid);
+            // It is important to print that, as long as isFocus is unreliable.
+            logger.info(
+                "Marking session from " + jingleSession.peerjid
+                + (jingleSession.isP2P ? " as P2P" : " as *not* P2P"));
+        }
+        if (jingleSession.isP2P) {
+            const role = this.room.getMemberRole(jingleSession.peerjid);
+            if ('moderator' === role) {
+                // Accept incoming P2P call
+                this._acceptP2PIncomingCall(jingleSession, jingleOffer);
+            } else {
+                // Reject incoming P2P call
+                this._rejectIncomingCall(jingleSession);
+            }
+        } else {
+            // Let the JitsiConference deal with the JVB session
+            super.onIncomingCall(jingleSession, jingleOffer, now);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     */
+    onMemberJoined (jid, nick, role, isHidden) {
+        super.onMemberJoined(jid, nick, role, isHidden);
+
+        this._startStopP2PSession();
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     */
+    onMemberLeft (jid) {
+        super.onMemberLeft(jid);
+
+        this._startStopP2PSession();
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     */
+    onRemoteTrackAdded (track) {
+        if (track.isP2P && !this.p2pEstablished) {
+            logger.info(
+                "Trying to add remote P2P track, when not in P2P - IGNORED");
+        } else if (!track.isP2P && this.p2pEstablished) {
+            logger.info(
+                "Trying to add remote JVB track, when in P2P - IGNORED");
+        } else {
+            super.onRemoteTrackAdded(track);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     * @override
+     */
+    onTransportInfo (jingleSession, transportInfo) {
+        if (this.peerToPeerSession === jingleSession) {
+            logger.info("Doing set transport-info");
+            this.peerToPeerSession.addIceCandidates(transportInfo);
+        }
+    }
+
+    /**
+     * Manually starts new P2P session (should be used only in the tests).
+     */
+    startPeer2PeerSession() {
+        const peers = this.getParticipants();
+        // Start peer to peer session
+        if (peers.length > 0) {
+            const peerJid = peers[0].getJid();
+            this._startPeer2PeerSession(peerJid);
+        } else {
+            logger.error("No participant to start the P2P session with !");
+        }
+    }
+
+    /**
+     * Manually stops the current P2P session (should be used only in the tests)
+     */
+    stopPeer2PeerSession() {
+        this._stopPeer2PeerSession();
+    }
+}
+
+/**
+ * This is a fake {@link ChatRoom} passed to the P2P {@link JingleSessionPC}
+ * in order to capture events emitted on it's event emitter (Jingle session uses
+ * chat room's emitter to send events).
+ */
+class FakeChatRoomLayer {
```

Comment:
Since this a fake ChatRoom, why not call it FakeChatRoom? That "layer" doesn't add much, does it?